pkg/sql/plan/function/builtin/unary/math.go

```diff
@@ -62,7 +62,37 @@ func Acos(vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
 }
 
 func Atan(vs []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
-	return math1(vs, proc, momath.Atan)
+	//If the vs's lenght is 1, just use the  function with one parameter
+	if len(vs) == 1 {
+		return math1(vs, proc, momath.Atan)
+	}
+	//Otherwise 
+	//First if one of the parameters is a null value
+	typ := vs[0].Typ.Oid.ToType() 
+	if vs[0].IsScalarNull() || vs[1].IsScalarNull(){
+		return proc.AllocScalarNullVector(typ), nil
+	}
+	
+	//Second if the vs is scalar
+	if vs[0].IsScalar() {
+		resultVector := proc.AllocScalarVector(types.Type{Oid: types.T_float64, Size: 8})
+		resultValues := make([]float64, 1)
+		vector.SetCol(resultVector, resultValues)
+		if err := momath.AtanWithTwoArg(vs[0], vs[1], resultVector); err != nil {
+			return nil, err
+		}
+		return resultVector, nil
+	}else{
+		vecLen := int64(vector.Length(vs[0]))
+		resultVector, err := proc.AllocVectorOfRows(types.T_float64.ToType(), vecLen, vs[0].Nsp)
+		if err != nil {
+			return nil, err
+		}
+		if err := momath.AtanWithTwoArg(vs[0], vs[1], resultVector); err != nil {
+			return nil, err
+		}
+		return resultVector, nil
+	}
```

Comment:
if you want to use your AtanWithTwoArg.
you can code like:
if vs[0].IsScalar() && vs[1].IsScalar() {
      // make scalar response and return
}

if vs[0].IsScalar() {
     // use vs[1] to set length and get resultVector
}
if vs[1].IsScalar() {
     // use vs[0] to set length and get resultVector
}
if err := momath.AtanWithTwoArg(vs[0], vs[1], resultVector); err != nil {
     return nil, err
}

Reply:
ok